### ISSUE TITLE
Buildozer requires a p4a version with AAB support.

### ISSUE THREAD

**Issue by flowerboyworld** [kivy/buildozer#1493]
### Versions

* Python: Python 3.9.7
* OS: MacOS 10.13.6
* Buildozer: 1.4.0

### Description

Tried to package an app. Changing the p4a.branch to develop does not seem to change anything.

### buildozer.spec
Command:
```sh
buildozer -v android debug
```

Spec file:
```
[app]

# (str) Title of your application
title = My Application

# (str) Package name
package.name = myapp

# (str) Package domain (needed for android/ios packaging)
package.domain = org.test

# (str) Source code where the main.py live
source.dir = .

# (list) Source files to include (let empty to include all the files)
source.include_exts = py,png,jpg,kv,atlas

# (list) List of inclusions using pattern matching
#source.include_patterns = assets/*,images/*.png

# (list) Source files to exclude (let empty to not exclude anything)
#source.exclude_exts = spec

# (list) List of directory to exclude (let empty to not exclude anything)
#source.exclude_dirs = tests, bin, venv

# (list) List of exclusions using pattern matching
# Do not prefix with './'
#source.exclude_patterns = license,images/*/*.jpg

# (str) Application versioning (method 1)
version = 0.1

# (str) Application versioning (method 2)
# version.regex = __version__ = ['"](.*)['"]
# version.filename = %(source.dir)s/main.py

# (list) Application requirements
# comma separated e.g. requirements = sqlite3,kivy
requirements = python3,kivy

# (str) Custom source folders for requirements
# Sets custom source for any requirements with recipes
# requirements.source.kivy = ../../kivy

# (str) Presplash of the application
#presplash.filename = %(source.dir)s/data/presplash.png

# (str) Icon of the application
#icon.filename = %(source.dir)s/data/icon.png

# (str) Supported orientation (one of landscape, sensorLandscape, portrait or all)
orientation = portrait

# (list) List of service to declare
#services = NAME:ENTRYPOINT_TO_PY,NAME2:ENTRYPOINT2_TO_PY

#
# OSX Specific
#

#
# author = © Copyright Info

# change the major version of python used by the app
osx.python_version = 3

# Kivy version to use
osx.kivy_version = 1.9.1

#
# Android specific
#

# (bool) Indicate if the application should be fullscreen or not
fullscreen = 0

# (string) Presplash background color (for android toolchain)
# Supported formats are: #RRGGBB #AARRGGBB or one of the following names:
# red, blue, green, black, white, gray, cyan, magenta, yellow, lightgray,
# darkgray, grey, lightgrey, darkgrey, aqua, fuchsia, lime, maroon, navy,
# olive, purple, silver, teal.
#android.presplash_color = #FFFFFF

# (string) Presplash animation using Lottie format.
# see https://lottiefiles.com/ for examples and https://airbnb.design/lottie/
# for general documentation.
# Lottie files can be created using various tools, like Adobe After Effect or Synfig.
#android.presplash_lottie = "path/to/lottie/file.json"

# (str) Adaptive icon of the application (used if Android API level is 26+ at runtime)
#icon.adaptive_foreground.filename = %(source.dir)s/data/icon_fg.png
#icon.adaptive_background.filename = %(source.dir)s/data/icon_bg.png

# (list) Permissions
#android.permissions = INTERNET

# (list) features (adds uses-feature -tags to manifest)
#android.features = android.hardware.usb.host

# (int) Target Android API, should be as high as possible.
#android.api = 27

# (int) Minimum API your APK / AAB will support.
#android.minapi = 21

# (int) Android SDK version to use
#android.sdk = 20

# (str) Android NDK version to use
#android.ndk = 23b

# (int) Android NDK API to use. This is the minimum API your app will support, it should usually match android.minapi.
#android.ndk_api = 21

# (bool) Use --private data storage (True) or --dir public storage (False)
#android.private_storage = True

# (str) Android NDK directory (if empty, it will be automatically downloaded.)
#android.ndk_path =

# (str) Android SDK directory (if empty, it will be automatically downloaded.)
#android.sdk_path =

# (str) ANT directory (if empty, it will be automatically downloaded.)
#android.ant_path =

# (bool) If True, then skip trying to update the Android sdk
# This can be useful to avoid excess Internet downloads or save time
# when an update is due and you just want to test/build your package
# android.skip_update = False

# (bool) If True, then automatically accept SDK license
# agreements. This is intended for automation only. If set to False,
# the default, you will be shown the license when first running
# buildozer.
# android.accept_sdk_license = False

# (str) Android entry point, default is ok for Kivy-based app
#android.entrypoint = org.kivy.android.PythonActivity

# (str) Full name including package path of the Java class that implements Android Activity
# use that parameter together with android.entrypoint to set custom Java class instead of PythonActivity
#android.activity_class_name = org.kivy.android.PythonActivity

# (str) Extra xml to write directly inside the <manifest> element of AndroidManifest.xml
# use that parameter to provide a filename from where to load your custom XML code
#android.extra_manifest_xml = ./src/android/extra_manifest.xml

# (str) Extra xml to write directly inside the <manifest><application> tag of AndroidManifest.xml
# use that parameter to provide a filename from where to load your custom XML arguments:
#android.extra_manifest_application_arguments = ./src/android/extra_manifest_application_arguments.xml

# (str) Full name including package path of the Java class that implements Python Service
# use that parameter to set custom Java class instead of PythonService
#android.service_class_name = org.kivy.android.PythonService

# (str) Android app theme, default is ok for Kivy-based app
# android.apptheme = "@android:style/Theme.NoTitleBar"

# (list) Pattern to whitelist for the whole project
#android.whitelist =

# (str) Path to a custom whitelist file
#android.whitelist_src =

# (str) Path to a custom blacklist file
#android.blacklist_src =

# (list) List of Java .jar files to add to the libs so that pyjnius can access
# their classes. Don't add jars that you do not need, since extra jars can slow
# down the build process. Allows wildcards matching, for example:
# OUYA-ODK/libs/*.jar
#android.add_jars = foo.jar,bar.jar,path/to/more/*.jar

# (list) List of Java files to add to the android project (can be java or a
# directory containing the files)
#android.add_src =

# (list) Android AAR archives to add
#android.add_aars =

# (list) Put these files or directories in the apk assets directory.
# Either form may be used, and assets need not be in 'source.include_exts'.
# 1) android.add_assets = source_asset_relative_path
# 2) android.add_assets = source_asset_path:destination_asset_relative_path
#android.add_assets =

# (list) Gradle dependencies to add
#android.gradle_dependencies =

# (bool) Enable AndroidX support. Enable when 'android.gradle_dependencies'
# contains an 'androidx' package, or any package from Kotlin source.
# android.enable_androidx requires android.api >= 28
#android.enable_androidx = False

# (list) add java compile options
# this can for example be necessary when importing certain java libraries using the 'android.gradle_dependencies' option
# see https://developer.android.com/studio/write/java8-support for further information
# android.add_compile_options = "sourceCompatibility = 1.8", "targetCompatibility = 1.8"

# (list) Gradle repositories to add {can be necessary for some android.gradle_dependencies}
# please enclose in double quotes 
# e.g. android.gradle_repositories = "maven { url 'https://kotlin.bintray.com/ktor' }"
#android.add_gradle_repositories =

# (list) packaging options to add 
# see https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.internal.dsl.PackagingOptions.html
# can be necessary to solve conflicts in gradle_dependencies
# please enclose in double quotes 
# e.g. android.add_packaging_options = "exclude 'META-INF/common.kotlin_module'", "exclude 'META-INF/*.kotlin_module'"
#android.add_packaging_options =

# (list) Java classes to add as activities to the manifest.
#android.add_activities = com.example.ExampleActivity

# (str) OUYA Console category. Should be one of GAME or APP
# If you leave this blank, OUYA support will not be enabled
#android.ouya.category = GAME

# (str) Filename of OUYA Console icon. It must be a 732x412 png image.
#android.ouya.icon.filename = %(source.dir)s/data/ouya_icon.png

# (str) XML file to include as an intent filters in <activity> tag
#android.manifest.intent_filters =

# (str) launchMode to set for the main activity
#android.manifest.launch_mode = standard

# (list) Android additional libraries to copy into libs/armeabi
#android.add_libs_armeabi = libs/android/*.so
#android.add_libs_armeabi_v7a = libs/android-v7/*.so
#android.add_libs_arm64_v8a = libs/android-v8/*.so
#android.add_libs_x86 = libs/android-x86/*.so
#android.add_libs_mips = libs/android-mips/*.so

# (bool) Indicate whether the screen should stay on
# Don't forget to add the WAKE_LOCK permission if you set this to True
#android.wakelock = False

# (list) Android application meta-data to set (key=value format)
#android.meta_data =

# (list) Android library project to add (will be added in the
# project.properties automatically.)
#android.library_references =

# (list) Android shared libraries which will be added to AndroidManifest.xml using <uses-library> tag
#android.uses_library =

# (str) Android logcat filters to use
#android.logcat_filters = *:S python:D

# (bool) Android logcat only display log for activity's pid
#android.logcat_pid_only = False

# (str) Android additional adb arguments
#android.adb_args = -H host.docker.internal

# (bool) Copy library instead of making a libpymodules.so
#android.copy_libs = 1

# (list) The Android archs to build for, choices: armeabi-v7a, arm64-v8a, x86, x86_64
# In past, was `android.arch` as we weren't supporting builds for multiple archs at the same time.
android.archs = arm64-v8a, armeabi-v7a

# (int) overrides automatic versionCode computation (used in build.gradle)
# this is not the same as app version and should only be edited if you know what you're doing
# android.numeric_version = 1

# (bool) enables Android auto backup feature (Android API >=23)
android.allow_backup = True

# (str) XML file for custom backup rules (see official auto backup documentation)
# android.backup_rules =

# (str) If you need to insert variables into your AndroidManifest.xml file,
# you can do so with the manifestPlaceholders property.
# This property takes a map of key-value pairs. (via a string)
# Usage example : android.manifest_placeholders = [myCustomUrl:\"org.kivy.customurl\"]
# android.manifest_placeholders = [:]

# (bool) disables the compilation of py to pyc/pyo files when packaging
# android.no-compile-pyo = True

# (str) The format used to package the app for release mode (aab or apk or aar).
# android.release_artifact = aab

# (str) The format used to package the app for debug mode (apk or aar).
# android.debug_artifact = apk

#
# Python for android (p4a) specific
#

# (str) python-for-android URL to use for checkout
#p4a.url =

# (str) python-for-android fork to use in case if p4a.url is not specified, defaults to upstream (kivy)
#p4a.fork = kivy

# (str) python-for-android branch to use, defaults to master
p4a.branch = develop

# (str) python-for-android specific commit to use, defaults to HEAD, must be within p4a.branch
#p4a.commit = HEAD

# (str) python-for-android git clone directory (if empty, it will be automatically cloned from github)
#p4a.source_dir =

# (str) The directory in which python-for-android should look for your own build recipes (if any)
#p4a.local_recipes =

# (str) Filename to the hook for p4a
#p4a.hook =

# (str) Bootstrap to use for android builds
# p4a.bootstrap = sdl2

# (int) port number to specify an explicit --port= p4a argument (eg for bootstrap flask)
#p4a.port =

# Control passing the --use-setup-py vs --ignore-setup-py to p4a
# "in the future" --use-setup-py is going to be the default behaviour in p4a, right now it is not
# Setting this to false will pass --ignore-setup-py, true will pass --use-setup-py
# NOTE: this is general setuptools integration, having pyproject.toml is enough, no need to generate
# setup.py if you're using Poetry, but you need to add "toml" to source.include_exts.
#p4a.setup_py = false

# (str) extra command line arguments to pass when invoking pythonforandroid.toolchain
#p4a.extra_args =


#
# iOS specific
#

# (str) Path to a custom kivy-ios folder
#ios.kivy_ios_dir = ../kivy-ios
# Alternately, specify the URL and branch of a git checkout:
ios.kivy_ios_url = https://github.com/kivy/kivy-ios
ios.kivy_ios_branch = master

# Another platform dependency: ios-deploy
# Uncomment to use a custom checkout
#ios.ios_deploy_dir = ../ios_deploy
# Or specify URL and branch
ios.ios_deploy_url = https://github.com/phonegap/ios-deploy
ios.ios_deploy_branch = 1.10.0

# (bool) Whether or not to sign the code
ios.codesign.allowed = false

# (str) Name of the certificate to use for signing the debug version
# Get a list of available identities: buildozer ios list_identities
#ios.codesign.debug = "iPhone Developer: <lastname> <firstname> (<hexstring>)"

# (str) The development team to use for signing the debug version
#ios.codesign.development_team.debug = <hexstring>

# (str) Name of the certificate to use for signing the release version
#ios.codesign.release = %(ios.codesign.debug)s

# (str) The development team to use for signing the release version
#ios.codesign.development_team.release = <hexstring>

# (str) URL pointing to .ipa file to be installed
# This option should be defined along with `display_image_url` and `full_size_image_url` options.
#ios.manifest.app_url =

# (str) URL pointing to an icon (57x57px) to be displayed during download
# This option should be defined along with `app_url` and `full_size_image_url` options.
#ios.manifest.display_image_url =

# (str) URL pointing to a large icon (512x512px) to be used by iTunes
# This option should be defined along with `app_url` and `display_image_url` options.
#ios.manifest.full_size_image_url =


[buildozer]

# (int) Log level (0 = error only, 1 = info, 2 = debug (with command output))
log_level = 2

# (int) Display warning if buildozer is run as root (0 = False, 1 = True)
warn_on_root = 1

# (str) Path to build artifact storage, absolute or relative to spec file
# build_dir = ./.buildozer

# (str) Path to build output (i.e. .apk, .aab, .ipa) storage
# bin_dir = ./bin

#    -----------------------------------------------------------------------------
#    List as sections
#
#    You can define all the "list" as [section:key].
#    Each line will be considered as a option to the list.
#    Let's take [app] / source.exclude_patterns.
#    Instead of doing:
#
#[app]
#source.exclude_patterns = license,data/audio/*.wav,data/images/original/*
#
#    This can be translated into:
#
#[app:source.exclude_patterns]
#license
#data/audio/*.wav
#data/images/original/*
#


#    -----------------------------------------------------------------------------
#    Profiles
#
#    You can extend section / key with a profile
#    For example, you want to deploy a demo version of your application without
#    HD content. You could first change the title to add "(demo)" in the name
#    and extend the excluded directories to remove the HD content.
#
#[app@demo]
#title = My Application (demo)
#
#[app:source.exclude_patterns@demo]
#images/hd/*
#
#    Then, invoke the command line with the "demo" profile:
#
#buildozer --profile demo android debug

```

### Logs

```
buildozer -v android debug
# Check configuration tokens
# Ensure build layout
# Check configuration tokens
# Read available permissions from api-versions.xml
# Preparing build
# Check requirements for android
# Run 'dpkg --version'
# Cwd None
/bin/sh: dpkg: command not found
# Search for Git (git)
#  -> found at /usr/bin/git
# Search for Cython (cython)
#  -> found at /Users/FlowerBoy/opt/anaconda3/bin/cython
# Search for Java compiler (javac)
#  -> found at /System/Library/Frameworks/JavaVM.framework/Versions/A/Commands/javac
# Search for Java keytool (keytool)
#  -> found at /System/Library/Frameworks/JavaVM.framework/Versions/A/Commands/keytool
# Install platform
# Run 'git config --get remote.origin.url'
# Cwd /Users/FlowerBoy/.buildozer/android/platform/python-for-android
https://github.com/kivy/python-for-android.git
# Run 'git branch -vv'
# Cwd /Users/FlowerBoy/.buildozer/android/platform/python-for-android
* master 17ec678c [origin/master] Merge pull request #2638 from misl6/release-2022.07.20
# Detected old url/branch (https://github.com/kivy/python-for-android.git/master), deleting...
# Run 'git clone -b develop --single-branch https://github.com/kivy/python-for-android.git python-for-android'
# Cwd /Users/FlowerBoy/.buildozer/android/platform
Cloning into 'python-for-android'...
# Run '/Users/FlowerBoy/opt/anaconda3/bin/python -m pip install -q  \'appdirs\' \'colorama>=0.3.3\' \'jinja2\' \'six\' \'enum34; python_version<"3.4"\' \'sh>=1.10; sys_platform!="nt"\' \'pep517<0.7.0\' \'toml\''
# Cwd None
# Apache ANT found at /Users/FlowerBoy/.buildozer/android/platform/apache-ant-1.9.4
# Android SDK found at /Users/FlowerBoy/.buildozer/android/platform/android-sdk
# Recommended android's NDK version by p4a is: 23b
# Android NDK found at /Users/FlowerBoy/.buildozer/android/platform/android-ndk-r23b
# Read available permissions from api-versions.xml
# Run '/Users/FlowerBoy/opt/anaconda3/bin/python -m pythonforandroid.toolchain aab -h --color=always --storage-dir="/Users/FlowerBoy/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a" --ndk-api=21 --ignore-setup-py --debug'
# Cwd /Users/FlowerBoy/.buildozer/android/platform/python-for-android
# This buildozer version requires a python-for-android version with AAB (Android App Bundle) support. Please update your pinned version accordingly.
# Check configuration tokens
# 

```

**Comment by misl6** [comment]
Hi @flowerboyworld!

Can you please try to use the latest `buildozer` `dev` version and see what happens on your side?
```
# latest dev version
# if you use a virtualenv, don't use the `--user` option
pip install --user https://github.com/kivy/buildozer/archive/master.zip
```

**Comment by flowerboyworld** [comment]
Hi

I think nothing changes, these are the logs:
```
FlowerBoy$ buildozer -v android debug
# Check configuration tokens
# Ensure build layout
# Check configuration tokens
# Read available permissions from api-versions.xml
# Preparing build
# Check requirements for android
# Search for Git (git)
#  -> found at /usr/bin/git
# Search for Cython (cython)
#  -> found at /Users/FlowerBoy/opt/anaconda3/bin/cython
# Search for Java compiler (javac)
#  -> found at /System/Library/Frameworks/JavaVM.framework/Versions/A/Commands/javac
# Search for Java keytool (keytool)
#  -> found at /System/Library/Frameworks/JavaVM.framework/Versions/A/Commands/keytool
# Install platform
# Run ['git', 'config', '--get', 'remote.origin.url']
# Cwd /Users/FlowerBoy/.buildozer/android/platform/python-for-android
https://github.com/kivy/python-for-android.git
# Run ['git', 'branch', '-vv']
# Cwd /Users/FlowerBoy/.buildozer/android/platform/python-for-android
* master 17ec678c [origin/master] Merge pull request #2638 from misl6/release-2022.07.20
# Detected old url/branch (https://github.com/kivy/python-for-android.git/master), deleting...
# Run ['git', 'clone', '-b', 'develop', '--single-branch', 'https://github.com/kivy/python-for-android.git', 'python-for-android']
# Cwd /Users/FlowerBoy/.buildozer/android/platform
Cloning into 'python-for-android'...
# Run ['/Users/FlowerBoy/opt/anaconda3/bin/python', '-m', 'pip', 'install', '-q', 'appdirs', 'colorama>=0.3.3', 'jinja2', 'six', 'enum34; python_version<"3.4"', 'sh>=1.10; sys_platform!="nt"', 'pep517<0.7.0', 'toml']
# Cwd None
# Apache ANT found at /Users/FlowerBoy/.buildozer/android/platform/apache-ant-1.9.4
# Android SDK found at /Users/FlowerBoy/.buildozer/android/platform/android-sdk
# Recommended android's NDK version by p4a is: 23b
# Android NDK found at /Users/FlowerBoy/.buildozer/android/platform/android-ndk-r23b
# Read available permissions from api-versions.xml
# Run ['/Users/FlowerBoy/opt/anaconda3/bin/python', '-m', 'pythonforandroid.toolchain', 'aab', '-h', '--color=always', '--storage-dir=/Users/FlowerBoy/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a', '--ndk-api=21', '--ignore-setup-py', '--debug']
# Cwd /Users/FlowerBoy/.buildozer/android/platform/python-for-android
# This buildozer version requires a python-for-android version with AAB (Android App Bundle) support. Please update your pinned version accordingly.
# Check configuration tokens
```

**Comment by flowerboyworld** [comment]
I get FileNotFoundError: [Errno 2] No such file or directory: 'brew'
```
[WARNING]: prerequisites.py is experimental and does not support all prerequisites yet.
[WARNING]: Please report any issues to the python-for-android issue tracker.
[ERROR]:   Prerequisite homebrew is not met
Traceback (most recent call last):
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/site-packages/pythonforandroid/toolchain.py", line 71, in <module>
    check_and_install_default_prerequisites()
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/site-packages/pythonforandroid/prerequisites.py", line 403, in check_and_install_default_prerequisites
    if not prerequisite.is_valid():
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/site-packages/pythonforandroid/prerequisites.py", line 18, in is_valid
    if self.checker():
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/site-packages/pythonforandroid/prerequisites.py", line 30, in checker
    return self.darwin_checker()
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/site-packages/pythonforandroid/prerequisites.py", line 299, in darwin_checker
    self._darwin_get_brew_formula_location_prefix("autoconf", installed=True)
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/site-packages/pythonforandroid/prerequisites.py", line 99, in _darwin_get_brew_formula_location_prefix
    p = subprocess.Popen(
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Users/FlowerBoy/opt/anaconda3/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'brew'
(base) Flowers-MBP:~ FlowerBoy$ FileNotFoundError: [Errno 2] No such file or directory: 'brew'
```

**Comment by misl6** [comment]
Ok, I guess we have an issue with the prerequisites helper, so please don't close this issue.
But, meanwhile, please install `brew` from: https://brew.sh, and then try again.

**Comment by flowerboyworld** [comment]
Thank you for your help
I have installed brew and now when I run buildozer it seems to be stuck; the last terminal output is 
```
# Run ['/Users/FlowerBoy/opt/anaconda3/bin/python', '-m', 'pythonforandroid.toolchain', 'aab', '-h', '--color=always', '--storage-dir=/Users/FlowerBoy/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a', '--ndk-api=21', '--ignore-setup-py', '--debug']
# Cwd /Users/FlowerBoy/.buildozer/android/platform/python-for-android
```

But nothing happens from there…

**Comment by misl6** [comment]
Ok, now just call again `/Users/FlowerBoy/opt/anaconda3/bin/python -m pythonforandroid.toolchain aab -h --color=always --storage-dir=/Users/FlowerBoy/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a --ndk-api=21 --ignore-setup-py --debug` manually and follow all the provided instructions.

When everything is set up, launch again the build via `buildozer`.

**Comment by flowerboyworld** [comment]
now I get a cython error, I suppose that's no longer buildozer territory …

```
Error compiling Cython file:
------------------------------------------------------------
...
import sys
import traceback

include "jnius_compat.pxi"
include "jni.pxi"
include "config.pxi"
^
------------------------------------------------------------

jnius/jnius.pyx:100:0: 'config.pxi' not found

Error compiling Cython file:
------------------------------------------------------------
...

include "jnius_compat.pxi"
include "jni.pxi"
include "config.pxi"

IF JNIUS_PLATFORM == "android":
  ^
------------------------------------------------------------

jnius/jnius.pyx:102:3: Compile-time name 'JNIUS_PLATFORM' not defined

Error compiling Cython file:
------------------------------------------------------------
...
include "jni.pxi"
include "config.pxi"

IF JNIUS_PLATFORM == "android":
    include "jnius_jvm_android.pxi"
ELIF JNIUS_PLATFORM == "win32":
    ^
------------------------------------------------------------

jnius/jnius.pyx:104:5: Compile-time name 'JNIUS_PLATFORM' not defined

Error compiling Cython file:
------------------------------------------------------------
...
include "config.pxi"
^
------------------------------------------------------------

jnius/jnius_jvm_dlopen.pxi:1:0: 'config.pxi' not found

Error compiling Cython file:
------------------------------------------------------------
...
    import jnius_config

    JAVA_LOCATION = get_java_setup()
    cdef str java_lib = JAVA_LOCATION.get_jnius_lib_location()

    IF JNIUS_PYTHON3:
      ^
------------------------------------------------------------

jnius/jnius_jvm_dlopen.pxi:57:7: Compile-time name 'JNIUS_PYTHON3' not defined

Error compiling Cython file:
------------------------------------------------------------
...

include "jnius_env.pxi"
include "jnius_utils.pxi"
include "jnius_conversion.pxi"
include "jnius_localref.pxi"
IF JNIUS_PYTHON3:
  ^
------------------------------------------------------------

jnius/jnius.pyx:113:3: Compile-time name 'JNIUS_PYTHON3' not defined
```

**Comment by misl6** [comment]
> now I get a cython error, I suppose that's no longer buildozer territory …
> 
> ```
> Error compiling Cython file:
> ------------------------------------------------------------
> ...
> import sys
> import traceback
> 
> include "jnius_compat.pxi"
> include "jni.pxi"
> include "config.pxi"
> ^
> ------------------------------------------------------------
> 
> jnius/jnius.pyx:100:0: 'config.pxi' not found
> 
> Error compiling Cython file:
> ------------------------------------------------------------
> ...
> 
> include "jnius_compat.pxi"
> include "jni.pxi"
> include "config.pxi"
> 
> IF JNIUS_PLATFORM == "android":
>   ^
> ------------------------------------------------------------
> 
> jnius/jnius.pyx:102:3: Compile-time name 'JNIUS_PLATFORM' not defined
> 
> Error compiling Cython file:
> ------------------------------------------------------------
> ...
> include "jni.pxi"
> include "config.pxi"
> 
> IF JNIUS_PLATFORM == "android":
>     include "jnius_jvm_android.pxi"
> ELIF JNIUS_PLATFORM == "win32":
>     ^
> ------------------------------------------------------------
> 
> jnius/jnius.pyx:104:5: Compile-time name 'JNIUS_PLATFORM' not defined
> 
> Error compiling Cython file:
> ------------------------------------------------------------
> ...
> include "config.pxi"
> ^
> ------------------------------------------------------------
> 
> jnius/jnius_jvm_dlopen.pxi:1:0: 'config.pxi' not found
> 
> Error compiling Cython file:
> ------------------------------------------------------------
> ...
>     import jnius_config
> 
>     JAVA_LOCATION = get_java_setup()
>     cdef str java_lib = JAVA_LOCATION.get_jnius_lib_location()
> 
>     IF JNIUS_PYTHON3:
>       ^
> ------------------------------------------------------------
> 
> jnius/jnius_jvm_dlopen.pxi:57:7: Compile-time name 'JNIUS_PYTHON3' not defined
> 
> Error compiling Cython file:
> ------------------------------------------------------------
> ...
> 
> include "jnius_env.pxi"
> include "jnius_utils.pxi"
> include "jnius_conversion.pxi"
> include "jnius_localref.pxi"
> IF JNIUS_PYTHON3:
>   ^
> ------------------------------------------------------------
> 
> jnius/jnius.pyx:113:3: Compile-time name 'JNIUS_PYTHON3' not defined
> ```

Yeah, that seems completely unrelated to the previous issue, can you try to completely delete your `.buildozer` folder and start again?

If you still encounter the Cython issue, feel free to join us on [Discord](https://chat.kivy.org), using one of the support channels, so we can better triage the issue.

**Comment by flowerboyworld** [comment]
by deleting my .buildozer folder, do you mean the package folder and install the package again?

**Comment by misl6** [comment]
> by deleting my .buildozer folder, do you mean the package folder and install the package again?

Nope, `buildozer` created a folder named `.buildozer` which is located into the same folder that stores your `buildozer.spec` file.

By deleting this folder, `buildozer` will start a new build from scratch.

**Comment by misl6** [comment]
Closed via #1494
